controllers/flinkcluster/flinkcluster_controller.go

```diff
@@ -46,18 +46,24 @@ type FlinkClusterReconciler struct {
 	Client        client.Client
 	Clientset     *kubernetes.Clientset
 	EventRecorder record.EventRecorder
+	Sharder       *Sharder
 }
 
 func NewReconciler(mgr manager.Manager) (*FlinkClusterReconciler, error) {
 	cs, err := kubernetes.NewForConfig(mgr.GetConfig())
 	if err != nil {
 		return nil, err
 	}
+	sh, err := NewSharderFromEnv()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Instead of relying only on env vars, we should support arguments whose values can be set with env vars.